### ISSUE TITLE
Colour contrast of points badges

### ISSUE THREAD

**Issue by mtthwhggns** [apluslms/a-plus#670]
As I've discussed here a lot, colour contrast is quite crucial for users with low vision, general readability and WCAG compliance.

The biggest problem are the points badges shown when a user has partial points, with white on orange. The white on green also fails to meet the requirements. 

Issue https://github.com/apluslms/a-plus/issues/568 was meant to address this, but was closed after the ability to change the colours was completed without changing all of them. 

We still have the problems. 

I implemented new colours in `assets/sass/colors/_functional.scss` which mirror the Bootstrap colour names. These were initialised with darker versions of their original colours, but weren't used by the badges (the badges are not part of Bootstrap). They caused the progress bars to appear in a different colour than the badges, which isn't ideal.

I tested replacing the badge colours with the darker colours to meet the required contrast ratio for small text, but it becomes much harder to tell them apart (dark green, dark orange, etc all look quite alike). 

![Screenshot 2020-08-27 at 13 34 10](https://user-images.githubusercontent.com/6674046/91432237-1c779d80-e86a-11ea-918c-3ddf0c6452f5.png)

This was where the suggestion to replace the badge with an icon, and beside the icon would be text in the normal colour, originated. The shape of the icon would provide a clear way to tell them apart, and colour could be used as a bonus. 

This was implemented in PR https://github.com/apluslms/a-plus/pull/646 however, since then, the discussion in https://github.com/apluslms/a-plus-design-system/issues/3 has stretched the requirements a little, and there was not time to finish it this summer.

If this cannot be solved by January's deployment time, I would recommend returning the Bootstrap `warning` colour to the bright orange, and add it as a known issue to the accessibility statement.

**Comment by raphendyr** [comment]
I would recommend to separate feedback points (used by graders) to a separate system. for those, there are "ok", "not ok", and "ignored" (currently missing, but needed by some assessments). For those, simple symbol and a color is fine.

Another set is points for submission, which is discussed in https://github.com/apluslms/a-plus-design-system/issues/3.

And finally, how exercises and modules represent the points (which is discussed in the above issue too).

Thus, I would a) change the warning color so every place in the A+ ios fixed, and also define set of points classes for assessment tools, .e.g., `aplus-points1-correct`, `aplus-points1-incorrect`, `aplus-points1-ignored` or so.

**Comment by Mankro** [comment]
We need a quick fix to this colour contrast issue before December 2020. The first version can be simple as long as it is "good enough" and passes the contrast requirement. #646 could be a better solution in the future, but it won't be finished now.

@jaguarfi Let's first try changing the badge colours so that there is enough contrast. @mtthwhggns already said that the darker colours look very alike. We will have to try to find acceptable colours. In addition, does increasing the font size reduce the contrast requirement? Is it feasible to increase the font size so that the colours can be more distinct? What if the shape of the badge is modified (e.g., from circle to square)?

**Comment by jaguarfi** [comment]
The `$font-size-base` was increased in the #690. Therefore, the `font-size` of the badges was increased as well.

**Comment by jaguarfi** [comment]
I also strongly support the #646 proposal.

**Comment by Mankro** [comment]
#699 was a quick fix. It changes the points badge colours as well as the progress bar colours so that the WAVE browser plugin accepts them and no longer warns about low contrast in the text of the badge.

#646 could be finished maybe in 2021 and it would replace these old points badges and progress bars with a new component.
